src/app/_services/api.service.ts

```diff
@@ -64,4 +67,62 @@ export class ApiService {
     return this.httpClient.get<any>(`${this.apiUrl}/accounts/reviews`);
   }
 
+
+  /*
+  * Group Methods
+  */
+
+  getGroups() {
+    return this.httpClient.get<any>(`${this.apiUrl}/groups`);
+  }
+
+  getGroup(groupId) {
+    return this.httpClient.get<any>(`${this.apiUrl}/groups/${groupId}`);
+  }
+
+  createGroup(group) {
+    return this.httpClient.post<any>(`${this.apiUrl}/groups`, group);
+  }
+
+
+  /*
+  * Post Methods
+  */
+
+  getPosts() {
+    return this.httpClient.get<any>(`${this.apiUrl}/posts`);
+  }
+
+  getPost(postId) {
+    return this.httpClient.get<any>(`${this.apiUrl}/posts/${postId}`);
+  }
+
+  getGroupPosts(groupId) {
+    return this.httpClient.get<any>(`${this.apiUrl}/groups/${groupId}/posts`);
+  }
+
+  createPost(post, groupId) {
+     return this.httpClient.post<any>(`${this.apiUrl}/groups/${groupId}/posts`, post);
+  }
+
+
+  /*
+  * Comment Methods
+  */
+
+  getComments() {
+    return this.httpClient.get<any>(`${this.apiUrl}/comments`);
+  }
+
+  getComment(commentId) {
+    return this.httpClient.get<any>(`${this.apiUrl}/comments/${commentId}`);
+  }
+
+  getPostComments(postId) {
+    return this.httpClient.get<any>(`${this.apiUrl}/posts/${postId}/comments`);
+  }
+
+  createComment(comment, postId) {
```

Comment:
again, reverse these arguments to match order in httpClient; postId, then comment object.